uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/impl/BaseKeyboardOperationTest.java

```diff
@@ -0,0 +1,211 @@
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import com.ait.lienzo.client.core.shape.Viewport;
+import com.ait.lienzo.client.core.types.Transform;
+import com.ait.lienzo.test.LienzoMockitoTestRunner;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.uberfire.ext.wires.core.grids.client.model.GridColumn;
+import org.uberfire.ext.wires.core.grids.client.model.GridData;
+import org.uberfire.ext.wires.core.grids.client.model.GridRow;
+import org.uberfire.ext.wires.core.grids.client.model.impl.BaseBounds;
+import org.uberfire.ext.wires.core.grids.client.widget.grid.GridWidget;
+import org.uberfire.ext.wires.core.grids.client.widget.grid.renderers.grids.GridRenderer;
+import org.uberfire.ext.wires.core.grids.client.widget.grid.renderers.grids.impl.BaseGridRendererHelper;
+import org.uberfire.ext.wires.core.grids.client.widget.layer.impl.DefaultGridLayer;
+
+import static com.google.gwt.event.dom.client.KeyCodes.KEY_LEFT;
+import static com.google.gwt.event.dom.client.KeyCodes.KEY_RIGHT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(LienzoMockitoTestRunner.class)
+public class BaseKeyboardOperationTest {
+
+    private static final double BOUNDS_WIDTH = 1000.0;
+    private static final double BOUNDS_HEIGHT = 1000.0;
+
+    @Mock
+    private DefaultGridLayer layer;
+
+    @Mock
+    private GridWidget gridWidget;
+
+    @Mock
+    private GridData gridData;
+
+    @Mock
+    private GridColumn gridColumn;
+
+    @Mock
+    private GridRow gridRow;
+
+    @Mock
+    private BaseGridRendererHelper baseGridRendererHelperMock;
+
+    @Mock
+    private BaseGridRendererHelper.RenderingInformation baseGridRendererInformationMock;
+
+    @Mock
+    private GridRenderer gridRenderer;
+
+    @Mock
+    private Viewport viewport;
+
+    private BaseKeyboardOperation baseKeyboardOperationSpy;
+    private int currentKeyCode = 0;
+
+    @Before
+    public void setup() {
+        when(gridWidget.getModel()).thenReturn(gridData);
+        when(gridWidget.getRendererHelper()).thenReturn(baseGridRendererHelperMock);
+        when(baseGridRendererHelperMock.getRenderingInformation()).thenReturn(baseGridRendererInformationMock);
+        when(gridData.getColumns()).thenReturn(Collections.singletonList(gridColumn));
+        when(gridColumn.getIndex()).thenReturn(0);
+        when(gridWidget.getRenderer()).thenReturn(gridRenderer);
+        when(layer.getViewport()).thenReturn(viewport);
+        when(layer.getVisibleBounds()).thenReturn(new BaseBounds(0, 0, BOUNDS_WIDTH, BOUNDS_HEIGHT));
+        when(viewport.getTransform()).thenReturn(new Transform());
```

Comment:
I think it'd be great to have an instance `mock` of `Transform` (instead of instantiate a `new` concrete instance)...

then....